x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/PackedValuesBlockHash.java

```diff
@@ -79,19 +76,19 @@ class AddWork extends LongLongBlockHash.AbstractAddBlock {
         final BatchEncoder[] encoders = new BatchEncoder[groups.size()];
         final int[] positionOffsets = new int[groups.size()];
         final int[] valueOffsets = new int[groups.size()];
-        final BytesRef[] scratches = new BytesRef[groups.size()];
+        final BytesRef scratch = new BytesRef();
+        final int[] loopedIndices = new int[groups.size()];
+        final int[] valueCounts = new int[groups.size()];
+        final int[] bytesStarts = new int[groups.size()];
```

Comment:
I think we'd be better of making an object for each group and have an array of those. But whatever works for you.

Reply:
++, will do.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/99992/commits/f41c6e83f1c03f92c021c1d1a8d3ded3bbd0d659, which is much cleaner.